site/content/en/docs/Installation/upgrading.md

```diff
@@ -0,0 +1,116 @@
+
+### Installation with install.yaml
+
+If you installed [Agones with install.yaml]({{< relref "_index.md#install-with-yaml" >}}), then you will need to delete
+the previous installation of Agones before upgrading to the new version, as we need to remove all of Agones before installing
+the new version.
+
+1. Start your maintenance window.
+1. Delete the current set of Fleets and/or GameServers in your cluster.
+1. Make sure to delete the same version of Agones that was previously installed, for example:
+   `kubectl delete -f https://raw.githubusercontent.com/googleforgames/agones/<old-release-version>/install/yaml/install.yaml`
+1. Install Agones [with install.yaml]({{< relref "_index.md#install-with-yaml" >}}).
+1. Deploy the same set of Fleets and/or GameServers back into the cluster.
+1. Run any other tests to ensure the Agones installation is working as expected.
+1. Close your maintenance window.
+7. Congratulations - you have now upgraded to a new version of Agones! 👍
+
+### Installation with Helm
+
+Helm features capabilities for upgrading to newer versions without having to delete the older version. For details on
+how to use Helm for upgrades, see the [helm upgrade](https://helm.sh/docs/helm/#helm-upgrade) documentation.
+
+Given the above, the steps for upgrade are simpler:
+
+1. Start your maintenance window.
+1. Run `helm upgrade` with the appropriate arguments, such a `--version`, for your specific upgrade
+1. Run any other tests to ensure the Agones installation is working as expected.
+1. Close your maintenance window.
+7. Congratulations - you have now upgraded to a new version of Agones! 👍
+
+## Upgrading Kubernetes
+
+The following are strategies for safely upgrading the underlying Kubernetes cluster from one version to another.
+They may require adjustment to your particular game architecture but should provide a solid foundation for updating your cluster safely.
+
+The recommended approach is to use [multiple clusters](#multiple-clusters-1), such that the upgrade can be tested
+gradually with production load and easily rolled back if need arises.
+
+### Multiple Clusters
+
+This process is very similar to the [Agones: Multiple Cluster](#multiple-clusters) approach above.
+
+Essentially, we want to do a [blue/green deployment](https://martinfowler.com/bliki/BlueGreenDeployment.html) to the new 
+version of Kubernetes.  This means that we can slowly migrate our game servers from the older version's cluster (blue) 
+to the new version's cluster (green), and ensure nothing surprising happens during this process.
+
+This also allows easy rollback to the previous infrastructure that we already know to be working in production,
+with minimal interruptions to player experience.
+
+The following are steps to implement this:
+
+1. Create a new cluster of the same size or smaller as the current cluster, with the new version of Kubernetes
+2. Install the same version of Agones on the new cluster, as you have on the previous cluster.
```

Comment:
This is useful if you want to separate the upgrade of k8s vs agones. But I imagine most folks will actually want to do them together (for minor upgrades) since agones will only be stress tested against a single patch version of k8s at a time.

Reply:
Added a note to this effect.  PTAL. I don't _think_ we need a combined K8s + Agones upgrade at the same time doc? Or do you think we do?

Reply:
I agree that we don't need another section (or doc) on doing them together. Just a note that as long as you are upgrading agones and either switching clusters or having maintenance that you might as well make sure that you are running the right k8s version underneath it.

Reply:
Done!

> Agones currently has only a [single supported Kubernetes version]({{< relref "_index.md#usage-requirements" >}}) for each version,
so it is recommended to do supported minor (e.g. 1.12.1 ➡ 1.13.2) Kubernetes version upgrades at the same time as matching Agones upgrades.